web/src/stream_list_sort.ts

```diff
@@ -23,9 +23,11 @@ let all_streams: number[] = [];
 // to avoid making left sidebar rendering a quadratic operation.
 let filter_out_inactives = false;
 
-export function get_streams(): (string | undefined)[] {
+export function get_streams(): string[] {
     const sorted_streams = all_streams.map((stream_id) =>
         sub_store.maybe_get_stream_name(stream_id),
+    ).filter(
+        (stream): stream is string => stream !== undefined
```

Comment:
The TypeScript compiler provides no safeguards at all against writing unsound type guards and type assertions (microsoft/TypeScript#33743), so we should avoid using those features when they aren’t necessary.

This function is easy to write using only safe features, such as `flatMap`:

```ts
    return all_streams.flatMap((stream_id) => {
        const stream_name = sub_store.maybe_get_stream_name(stream_id);
        return stream_name === undefined ? [] : [stream_name];
    });
```

or boring loops:

```ts
    const stream_names = [];
    for (const stream_id of all_streams) {
        const stream_name = sub_store.maybe_get_stream_name(stream_id);
        if (stream_name !== undefined) {
            stream_names.push(stream_name);
        }
    }
    return stream_names;
```

Reply:
Ah perfect, thanks!